Mathlib/Topology/ProperMap.lean

```diff
@@ -136,6 +137,60 @@ lemma IsProperMap.ultrafilter_le_nhds_of_tendsto (h : IsProperMap f) ⦃𝒰 : U
     (hy : Tendsto f 𝒰 (𝓝 y)) : ∃ x, f x = y ∧ 𝒰 ≤ 𝓝 x :=
   (isProperMap_iff_ultrafilter.mp h).2 hy
 
+/-- The composition of two proper maps is proper. -/
+lemma IsProperMap.comp (hf : IsProperMap f) (hg : IsProperMap g) :
+    IsProperMap (g ∘ f) := by
+  refine ⟨by continuity, fun ℱ z h ↦ ?_⟩
+  rw [MapClusterPt, ← map_map] at h
+  rcases hg.clusterPt_of_mapClusterPt h with ⟨y, hy1, hy2⟩
+  rcases hf.clusterPt_of_mapClusterPt hy2 with ⟨x, hx1, hx2⟩
+  use x
+  rw [← hy1, ← hx1]
+  exact ⟨rfl, by assumption⟩
+
+/-- If the composition of two continuous functions `g ∘ f` is proper and `f` is surjective,
+then `g` is proper. -/
+lemma isProperMap_of_comp_of_surj (hf : Continuous f)
+    (hg : Continuous g) (hgf : IsProperMap (g ∘ f)) (f_surj : f.Surjective) : IsProperMap g := by
+  refine ⟨hg, fun ℱ z h ↦ ?_⟩
+  rcases hgf with ⟨_, h'⟩
```

Comment:
This is a very minor stylistic comment: notice that the `_` still introduces an (unnamed) hypothesis, which clutters the goal view and may still be used by tactics later. Instead, you could do the following to completely discard this hypothesis:
```suggestion
  rcases hgf with ⟨-, h'⟩
```

Another solution, maybe better, would be to directly use `hgf.clusterPt_of_mapClusterPt` where you use `h'`.

This applies to other occurrences of the same pattern below.